src/util/bns.ts

```diff
@@ -0,0 +1,7 @@
+export function isValidBNSName(name: string): boolean {
+  if (name) {
+    const dotIndex = name.indexOf(".");
+    return dotIndex > 0 && dotIndex < name.length - 1; // dot comes in between
```

Comment:
I'm not 100% sure about the BNS format, but `"a.b."` would pass this filter, and I don't think it's a valid name. Looking at BNS hub, it says _Only lowercase letters, numbers, hyphens, and underscores are allowed_ when searching for a domain, and the smart contract seem to have a similar logic. But then in BNS hub I see emojii, so maybe there's something else going on.

Anyway, IIUC this is just to avoid "wasting" queries to BNS, so it's not an issue if we have false negatives. But we could improve this a little bit.